docker_cli/_image.py

```diff
@@ -0,0 +1,471 @@
+        return the associated Image instance.
+
+        Parameters
+        ----------
+        tag : str
+            A name for the image.
+        context : os.PathLike, optional
+            The build context. Defaults to ".".
+        dockerfile : os.PathLike, optional
+            The path of the Dockerfile to build, relative to the `context`
+            directory. Defaults to None.
+        stdout : io.TextIOBase or special value, optional
+            For a description of valid values, see :func:`subprocess.run`.
+        stderr : io.TextIOBase or special value, optional
+            For a description of valid values, see :func:`subprocess.run`.
+        network : str, optional
+            The name of the network. Defaults to "host".
+        no-cache : bool, optional
+            A boolean designating whether or not the docker build should use
+            the cache.
+
+        Returns
+        -------
+        Image
+            The created image.
+
+        Raises
+        -------
+        DockerBuildError
+            If the docker build command fails.
+        ValueError
+            If both `dockerfile` and `dockerfile_string` are defined.
+        """
+        ...
+
+    @overload
+    @classmethod
+    def build(
+        cls: Type[Self],
+        tag: str,
+        *,
+        context: Union[str, os.PathLike[str]],
+        dockerfile_string: str,
+        stdout: Optional[io.TextIOBase],
+        stderr: Optional[io.TextIOBase],
+        network: str,
+        no_cache: bool
+    ) -> Self:
+        """
+        Builds a new image from a string in dockerfile syntax.
+
+        Parameters
+        ----------
+        tag : str
+            A name for the image.
+        context : os.PathLike, optional
+            The build context. Defaults to ".".
+        dockerfile_string : str
+            A Dockerfile-formatted string.
+        stdout : io.TextIOBase or special value, optional
+            For a description of valid values, see :func:`subprocess.run`.
+        stderr : io.TextIOBase or special value, optional
+            For a description of valid values, see :func:`subprocess.run`.
+        network : str, optional
+            The name of the network. Defaults to "host".
+        no-cache : bool, optional
+            A boolean designating whether or not the docker build should use
+            the cache.
+
+        Returns
+        -------
+        Image
+            The created Image.
+
+        Raises
+        -------
+        DockerBuildError
+            If the docker build command fails.
+        ValueError
+            If both `dockerfile` and `dockerfile_string` are defined.
+        """
+        ...
+
+    @classmethod
+    def build(
+        cls,
+        tag,
+        *,
+        context=".",
+        dockerfile=None,
+        dockerfile_string=None,
+        stdout=None,
+        stderr=None,
+        network="host",
+        no_cache=True
```

Comment:
Let's default `no_cache` to False, matching docker's own default behavior. Disabling the cache is something that the user should need to explicitly ask for.

```suggestion
        no_cache=False
```